WMF Framework/Remote Notifications/Model/RemoteNotification+CoreDataClass.swift

```diff
@@ -15,6 +15,18 @@ import CoreData
     }
 }
 
+public struct RemoteNotificationState: OptionSet {
+    public let rawValue: Int16
+
+    public static let wasSeen = RemoteNotificationState(rawValue: 1 << 0)
+    public static let wasRead = RemoteNotificationState(rawValue: 1 << 1)
+    public static let isExcluded = RemoteNotificationState(rawValue: 1 << 2)
+
```

Comment:
When user sees a notification, the state will be asynchronously set to **wasSeen** ( call A: https://github.com/wikimedia/wikipedia-ios/pull/2663/files#diff-82cb2edf849f93541737d9c5c5ff2cfeR1677), if user reads through, the state will be asynchronously set to **wasRead** (call B: https://github.com/wikimedia/wikipedia-ios/pull/2663/files#diff-82cb2edf849f93541737d9c5c5ff2cfeR1709 or https://github.com/wikimedia/wikipedia-ios/pull/2663/files#diff-82cb2edf849f93541737d9c5c5ff2cfeR1712 ). Given that both operations happen asynchronously, I think there could be a situation where call B finishes before call A, which would result in us not marking the notification as read on the server (and keeping It around for no reason)

Reply:
do both changes happen on the private queue of the moc?

Reply:
yes, inside `perform`

Reply:
that should ensure the calls happen serially